ipatests/test_integration/test_backup_and_restore.py

```diff
@@ -421,3 +421,49 @@ class TestBackupReinstallRestoreWithKRA(BaseBackupAndRestoreWithKRA):
     def test_full_backup_reinstall_restore_with_vault(self):
         """backup, uninstall, reinstall, restore"""
         self._full_backup_restore_with_vault(reinstall=True)
+
+
+class TestBackupNotFailesSilently(IntegrationTest):
+    """Before the fix, when ipa-backup was called for the first time,
+    the LDAP database exported to
+    /var/lib/dirsrv/slapd-<instance>/ldif/<instance>-userRoot.ldif.
+    db2ldif is called for this and it runs under root, hence files
+    were owned by root.
+
+    When ipa-backup called the next time, the db2ldif fails,
+    because the tool does not have permissions to write to the ldif
+    file which was owned by root (instead of dirsrv)
+
+    This test check if files are owned by dirsrv and db2ldif doesn't
+    fails
+
+    related ticket: https://pagure.io/freeipa/issue/7010 """
+
+    def test_backup_not_failes_silently(self):
+        """backup, uninstall, restore, backup"""
+        tasks.install_master(self.master)
+        backup_path = backup(self.master)
+
+        self.master.run_command(['ipa-server-install',
+                                 '--uninstall',
+                                 '-U'])
+
+        dirman_password = self.master.config.dirman_password
+        self.master.run_command(['ipa-restore', backup_path],
+                                stdin_text=dirman_password + '\nyes')
+
+        dashed_domain = self.domain.realm.replace(".", '-')
+        arg = ['ls', '-ldZ',
+               '/var/lib/dirsrv/slapd-' + dashed_domain + '/ldif']
+        cmd = self.master.run_command(arg)
+        assert 'dirsrv dirsrv' in cmd.stdout_text
+
+        arg = ['ls', '-lZ',
+               '/var/lib/dirsrv/slapd-' + dashed_domain + '/ldif/']
+        cmd = self.master.run_command(arg)
+        assert 'dirsrv dirsrv' in cmd.stdout_text
+
+        cmd = self.master.run_command(['ipa-backup', '-d'])
+        unexp_str = "CRITICAL: db2ldif failed:"
+        assert cmd.returncode == 0
```

Comment:
In case of an error, the return code should not be 0. If it's 0, then it's a bug in ipa-backup command.

Reply:
Command is suppose to be success here. I am just checking if `unexp_str` doesn't appear in `stdout` as it was appearing before fix.